xonsh/built_ins.py

```diff
@@ -613,6 +613,10 @@ def run_subproc(cmds, captured=True):
             'obj': prev_proc,
             'bg': background
         })
+    try:
+        builtins.__xonsh_shell__.settitle()
+    except AttributeError:
+        pass
```

Comment:
Ahh ok.  Yeah, that makes more sense.  Some of the tests mock the builtins initialization. I think that code is in `tests/tools.py`. As I recall, the `__xonsh_shell__` came later. So it might not be mocked.